hack/boilerplate/boilerplate.py

```diff
@@ -181,16 +185,95 @@ def get_regexs():
     regexs["go_build_constraints"] = re.compile(r"^(// \+build.*\n)+\n", re.MULTILINE)
     # strip #!.* from shell scripts
     regexs["shebang"] = re.compile(r"^(#!.*\n)\n*", re.MULTILINE)
+    # Search for generated files
+    regexs["generated"] = re.compile( 'DO NOT EDIT' )
     return regexs
 
+
+def is_generated_files(filename, regexs):
+    for d in skipped_ungenerated_files:
+ 	if d in filename:
+            return False
+    
+    try:
+        f = open(filename, 'r')
+    except Exception as exc:
+        print("Unable to open %s: %s" % (filename, exc), file=verbose_out)
+        return False
+
+    data = f.read()
+    f.close()
+
+    p = regexs["generated"]
+    found = p.search(data)
+
+    if found:
+        return True
+    else:
+        return False
+
+
+def generated_file_passes(filename, refs, regexs):
```

Comment:
this needs docs. It's not clear what this func does.

Reply:
OK

Reply:
What is refs? What are the regexs? What is returned? What are the side-effects?